src/marqo/errors.py

```diff
@@ -165,3 +165,27 @@ class BackendTimeoutError(InternalError):
     def __init__(self, message: str,) -> None:
         self.message = f"Timeout error communicating with Marqo: {message}"
 
+
+class MarqoIndexNotReadyError(__InvalidRequestError):
+    """Error when Marqo index is not ready"""
+    code = "index_not_ready_cloud"
+    status_code = HTTPStatus.NOT_FOUND
+
+    def __init__(self, index_name: str,) -> None:
+        self.message = f"The Python client could not resolve the endpoint for the index name {index_name}. " \
+                       f"This could be due to the index is still in the process of being created," \
+                       f" or that the client's cache has not yet been updated.\n" \
+                       f"- Please try again in a couple of minutes, " \
+                       f"or you can query the index status with get_status function to see when index is ready.\n" \
```

Comment:
`mq.('my-index').get_status()`

Reply:
- Consider subclassing this, because it's a client specific, rather than HTTP errror: https://github.com/marqo-ai/py-marqo/blob/ab2c3de94f02ad0ef8440da743f6e18600f701a1/src/marqo/errors.py#L6